tests/installation/welcome.pm

```diff
@@ -18,36 +18,44 @@ use testapi;
 use utils 'ensure_fullscreen';
 
 sub run() {
-    my @welcome_tags = [qw(inst-welcome inst-welcome-confirm-self-update-server)];
-    ensure_fullscreen;
-    my $bootup_timeout = 500;
-    if (get_var("BETA")) {
-        assert_screen "inst-betawarning", $bootup_timeout;
-        send_key "ret";
-        assert_screen @welcome_tags, 10;
-    }
-    else {
-        assert_screen @welcome_tags, $bootup_timeout;
-    }
-    if (match_has_tag('inst-welcome-confirm-self-update-server')) {
-        wait_screen_change { send_key $cmd{ok} };
-        assert_screen 'inst-welcome';
-    }
+    while (1) {
```

Comment:
Why do you use cycle here? Do you expect some needle to match more than once?
Why not make it simpler by:
 - assert_screen scc-invalid-url if SCC_URL_INVALID
 - assert_screen inst-betawarning if BETA
 - assert_screen inst-welcome-confirm-self-update if UPDATE
 - assert_screen inst_welcome